app/rdf/query-latest-cube-iri.ts

```diff
@@ -17,27 +17,31 @@ SELECT ?iri WHERE {
       ?oldIri schema:creativeWorkStatus ?oldStatus .
       FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status IN (?oldStatus, <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>))
     }
+    ORDER BY DESC(?version)
   } UNION {
     {
-      # Non-versioned cube.
-      SELECT ?iri WHERE {
-        VALUES ?iri { <${cubeIri}> }
-        ?iri cube:observationConstraint ?shape .
+      # Version history of a cube.
+      SELECT ?iri ?status ?version WHERE {
+        VALUES ?versionHistory { <${cubeIri}> }
+        ?versionHistory schema:hasPart ?iri .
+        ?iri schema:version ?version .
         ?iri schema:creativeWorkStatus ?status .
-        FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status = <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>)      
+        FILTER(NOT EXISTS { ?iri schema:expires ?expires . })
```

Comment:
@bprusinowski this subquery is also responding in versioned cases - this should be prevented to not interfere and spill over Draft in versioned cases.

i.e. adding `FILTER NOT EXISTS { ?versionHistory schema:hasPart ?iri . }` should do the trick.

Reply:
@bprusinowski sorry - I've overlooked the new order of the 3 subqueries...